### DISCUSSION TITLE
UWB bias neural network fitting embedded code

### DISCUSSION THREAD

**Discussion by CalvinKoung** [bitcraze/discussions#1364] [Q&A]
Hi, I saw the blog post about Robust Kalman Filter for Ultra-wideband Localization. The paper mention neural netwok is KristofferRichardsson(Bitcraze) help to guidance embedded.  Can I know the detail ? I want put my neural network biasac and weight to add in  that I can robust in my environment. thanks

**Comment by knmcguire** [comment]
Hi! I'm assuming that you are referring to this blogpost right? https://www.bitcraze.io/2021/06/robust-kalman-filter-for-ultra-wideband-localization/

Kristoffer is on a hiatus from Bitcraze so he is not able to help out. You'll need to look into the code yourself, so I advise to start with the robustTDOA parameter how it is implemented into the code https://github.com/bitcraze/crazyflie-firmware/blob/a25e8fb2254f6d5be54ad57566b3b846a97de93b/src/modules/src/estimator/estimator_kalman.c#L520

Also this is the PR that implemented this work: https://github.com/bitcraze/crazyflie-firmware/pull/707

**Reply by CalvinKoung** [reply]
OK, I got it. tks

**Comment by CalvinKoung** [comment]
But where can I modify the robustTwr parameter ? tks

**Reply by knmcguire** [reply]
You mean the value of it? You can change the value from the cfclient in the parameter tab: https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/parameters_tab/

Or you can write your own script and do it through the cfclient https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/user-guides/python_api/#parameters-1
